src/Stryker.Core/Stryker.Core/Baseline/DiskBaselineProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Stryker.Core.Logging;
+using Stryker.Core.Options;
+using Stryker.Core.Reporters.Json;
+using System.IO;
+using System.IO.Abstractions;
+using System.Threading.Tasks;
+
+namespace Stryker.Core.Baseline
+{
+    public class DiskBaselineProvider : IBaselineProvider
+    {
+        private readonly StrykerOptions _options;
+        private readonly IFileSystem _fileSystem;
+        private readonly ILogger<DiskBaselineProvider> _logger;
+
+        public DiskBaselineProvider(StrykerOptions options, IFileSystem fileSystem = null)
+        {
+            _options = options;
+            _fileSystem = fileSystem ?? new FileSystem();
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<DiskBaselineProvider>();
+        }
+
+
+        public async Task<JsonReport> Load(string version)
+        {
+            var reportPath = Path.Combine(_options.BaselineOutputPath, version, "stryker-report.json");
+            try
+            {
+                using StreamReader inputReader = _fileSystem.File.OpenText(reportPath);
+
+                var reportJson = await inputReader.ReadToEndAsync();
+
+                return JsonConvert.DeserializeObject<JsonReport>(reportJson);
+            } 
+            catch(FileNotFoundException)
```

Comment:
The baseline not existing is not an unexpected failure situation, it probably happens quite often. Use File.Exists instead of a try-catch

Reply:
Also, the first time you run dashboard compare the directory Baselines\dashboard-compare\<version> itself does not exist yet, so a catch on FileNotFoundException is not sufficient.